src/EVM/Expr.hs

```diff
@@ -1182,6 +1188,7 @@ simplifyNoLitToKeccak e = untilFixpoint (mapExpr go) e
       | otherwise = shr a v
 
     -- Bitwise AND & OR. These MUST preserve bitwise equivalence
+    go (And (Lit a) (Lit b)) = Lit (a .&. b)
```

Comment:
This should not be necessary, no? This simplification should happen in `EVM.Expr.and a b`.

Reply:
Oops, good point. OK, I'll write test cases :)